extensions-core/druid-lookups/src/test/java/io/druid/server/lookup/PollingLookupTest.java

```diff
@@ -0,0 +1,211 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.server.lookup;
+
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import com.google.common.base.Function;
+import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Maps;
+import com.google.common.collect.Sets;
+import com.metamx.common.ISE;
+import io.druid.query.lookup.LookupExtractor;
+import io.druid.server.lookup.cache.polling.OffHeapPollingCache;
+import io.druid.server.lookup.cache.polling.OnHeapPollingCache;
+import io.druid.server.lookup.cache.polling.PollingCacheFactory;
+import org.joda.time.Period;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import javax.annotation.Nullable;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+@RunWith(Parameterized.class)
+public class PollingLookupTest
+{
+  private static final Map<String, String> firstLookupMap = ImmutableMap.of(
+      "foo", "bar",
+      "bad", "bar",
+      "how about that", "foo",
+      "empty string", ""
+  );
+
+  private static final Map<String, String> secondLookupMap = ImmutableMap.of(
+      "new-foo", "new-bar",
+      "new-bad", "new-bar"
+  );
+
+  private static final int POLL_PERIOD = 100;
+
+  @JsonTypeName("mock")
+  private static class MockDataFetcher implements DataFetcher
+  {
+    private int callNumber = 0;
+    @Override
+    public Map fetchAll()
+    {
+      if (callNumber == 0) {callNumber +=1; return firstLookupMap;}
+      return secondLookupMap;
+    }
+
+    @Nullable
+    @Override
+    public Object fetch(Object key)
+    {
+      return null;
+    }
+
+    @Override
+    public Map fetch(Iterable keys)
+    {
+      return null;
+    }
+
+    @Override
+    public List reverseFetchKeys(Object value)
+    {
+      return null;
+    }
+
+    @Override
+    public boolean equals(Object obj)
+    {
+      return obj instanceof MockDataFetcher;
+    }
+  }
+
+  @Parameterized.Parameters
+  public static Collection<Object[]> inputData()
```

Comment:
just FYI, this technically only has to be `Iterable<Object[]>` which means you can feed tests with more things than would fit in test memory if you materialized all the parameters at once.